### PR TITLE
Invisible Light Source

### LINKED ISSUE #21: Invisible Light Source
Invisible Light Source

Serves as a light source, similar to dash refills, lamps, and Madeline. No visible sprite or anything, though.

Requested by LethargicDoggo in collabcord

### PR DESCRIPTION
**Pull request by WoofWoofDoggo**
Adds an entity called `Light Source (Spring Collab 2020)` to Ahorn, and the game. Acts as an invisible light source, with configurable radius, color, and opacity.

Also contains tooltips.

Fixes #21

### CODE REVIEW

File: Entities/InvisibleLightSource.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            radius = data.Float("radius", 48f);
+            color = ColorHelper.GetColor(data.Attr("color", "White"));
+
+            bloom = new BloomPoint(alpha, radius);
+            light = new VertexLight(color, alpha, data.Int("startFade", 24), data.Int("endFade", 48));
+
+            bloom.Visible = true;
+            light.Visible = true;
+
+            base.Add(bloom);
```
Comment (0x0ade) [inline Entities/InvisibleLightSource.cs]:
```suggestion
            Add(bloom);
```

File: Entities/InvisibleLightSource.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            base.Add(light);
```
Comment (0x0ade) [inline Entities/InvisibleLightSource.cs]:
```suggestion
            Add(light);
```

File: Entities/InvisibleLightSource.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+        private float radius;
+
+        private Color color;
+    }
+
+    // Cruor made this
+    class ColorHelper {
+        public static Color GetColor(string color) {
+            foreach(PropertyInfo c in colorProps) {
```
Comment (0x0ade) [inline Entities/InvisibleLightSource.cs]:
```suggestion
            foreach (PropertyInfo c in colorProps) {
```

**Comment by Cruor** [comment]
Missing a dropdown for light color, and I would still prefer if the entity was drawn a bit see through in Ahorn.

**Comment by 0x0ade** [comment]
Praise GitHub's review suggestion thing.

File: Ahorn/entities/invisibleLightSource.jl [hunk 4]
```diff
@@ -1,23 +1,34 @@
...
 
+Ahorn.editingOptions(entity::InvisibleLightSource) = Dict{String,Any}(
+	"color" => colors
+)
+
 function Ahorn.selection(entity::InvisibleLightSource)
 	x, y = Ahorn.position(entity)
 
-	return Ahorn.Rectangle(x, y, 9, 12)
+	return Ahorn.Rectangle(x, y, 7, 8)
```
Comment (Cruor) [inline Ahorn/entities/invisibleLightSource.jl]: Should have offset for drawing

File: Ahorn/entities/invisibleLightSource.jl [hunk 5]
```diff
@@ -1,23 +1,34 @@
 ﻿module SpringCollab2020InvisibleLightSource
+
 using ..Ahorn, Maple
+
 @mapdef Entity "SpringCollab2020/invisibleLightSource" InvisibleLightSource(x::Integer, y::Integer, alpha::Number=1, radius::Number=48, startFade::Number=24, endFade::Number=48, color::String="White")
 
+#Ahorn.drawImage(ctx, "objects/hanginglamp", x + 1, y - 14)
```
Comment (Cruor) [inline Ahorn/entities/invisibleLightSource.jl]: Please remove

**Pull request merged**